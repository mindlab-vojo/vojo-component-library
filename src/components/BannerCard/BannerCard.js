import React from 'react'
import PropTypes from 'prop-types'

import * as BannerCard from './BannerCard.style'

import ThemeConsumer from '../../style/ThemeManager/ThemeProvider'
import { Card } from '../Card/Card.style'
import Typography from '../Typography'

const BannerCardComponent = ({
  description,
  id,
  onClickCard,
  title,
}) => {

  const renderComponent = (theme) => (
    <Card 
      hasPointer
      id={`BannerCard-${id}`}
      padding="15px 8px" 
      onClick={onClickCard}>
      <BannerCard.Wrapper>
        <BannerCard.LeftContent>
          <BannerCard.Image
            theme={theme}>
            <Typography
              color={theme.colors.darkBlueColor}
              fontSize="18px"
              fontWeight="bolder">
              ?
            </Typography>
          </BannerCard.Image>
        </BannerCard.LeftContent>
        <BannerCard.RightContent>
          <BannerCard.Description>
            <Typography
              fontSize="12px">
              {description}
            </Typography>
          </BannerCard.Description>
          <BannerCard.Title>
            <Typography
              color={theme.colors.darkBlueColor}
              fontSize="18px"
              fontWeight="bolder">
              {title}
            </Typography>
          </BannerCard.Title>
        </BannerCard.RightContent>
      </BannerCard.Wrapper>
    </Card>
  )

  return (
    <ThemeConsumer>
      {({ theme }) => renderComponent(theme)}
    </ThemeConsumer>
  )
}

BannerCardComponent.propTypes = {
  id: PropTypes.string,
  description: PropTypes.string,
  onClickCard: PropTypes.func,
  title: PropTypes.string,
}

export default BannerCardComponent