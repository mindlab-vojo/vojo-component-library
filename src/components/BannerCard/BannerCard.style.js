import styled from 'styled-components'

export const Wrapper = styled.div`
  display: flex;
  width: 100%;
`
export const Image = styled.div`
  align-items: center;
  background-color: ${props => props.theme.colors.lightYellowColor};
  border-radius: 500px;
  display: flex;
  height: 56.93px;
  justify-content:center;
  width: 56.93px;
`

export const LeftContent = styled.div`
  align-items: center;
  display: flex;
  justify-content:center;
` 

export const RightContent = styled.div`
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content:center;
  margin-left: 9px;
` 

export const Title = styled.div`
  width: 100%;
` 

export const Description = styled.div`
  margin-bottom: 5px;
  width: 100%;
`
