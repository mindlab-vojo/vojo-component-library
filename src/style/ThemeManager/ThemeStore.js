import { vojo } from './../themes'

class ThemeStore {

  constructor(theme) {
    this.defaultTheme = vojo
    this.theme = theme ? theme : this.defaultTheme
  }

  configure = ({ theme }) =>  {
    if (theme) this.configureTheme(theme)
    else this.configureTheme(this.defaultTheme)
  }

  configureTheme = (theme) => {
    this.theme = theme
  }

  modify = (modifiedProps) => {
    const currentTheme = this.theme
    return { currentTheme, modifiedProps }
  }

  getTheme = () => this.theme

}

export default ThemeStore