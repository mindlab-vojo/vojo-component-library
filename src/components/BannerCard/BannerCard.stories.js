import React from 'react';
import { withKnobs, text } from '@storybook/addon-knobs';
import { withA11y } from '@storybook/addon-a11y';
import { action } from '@storybook/addon-actions'

import BannerCard from '.'; 
import { ThemeProvider } from "../../../src";
import categories from '../../../.storybook/categories';

export const bannerCard = () => {  
  const id = text('id', 'card-id')
  const onClickCard = action('onClickCard')
  const title = text('title', 'Entregador Mágico')
  const description = text('description', 'Entregador Mágico')

  return (
    <BannerCard
      description={description}
      id={id}
      onClickCard={onClickCard}
      title={title}
    />
  )
}

export default {
  title: categories.Cards,
  name: 'BannerCard',
  component: bannerCard,
  decorators: [
    withKnobs, 
    withA11y, 
    (story) => <ThemeProvider>{ story() }</ThemeProvider>,
    (story) => <div style={{ display: "flex" }}>{ story() }</div>,
  ],
}
  