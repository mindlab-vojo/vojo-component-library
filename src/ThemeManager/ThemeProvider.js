import React, { Component, createContext } from 'react'
import PropTypes from 'prop-types'

import ThemeStore from './ThemeStore';

// Provider and Consumer instances
const { Provider, Consumer } = createContext();

class ThemeProvider extends Component {
  static propTypes = {
    children: PropTypes.node,
    theme: PropTypes.string
  }

  state = {
    themeConfig: new ThemeStore()
  }

  componentDidMount() {
    if (this.props.theme) this.setTheme()
  }

  componentDidUpdate(prevProps) {
    if (this.props.theme !== prevProps.theme) this.setTheme()
  }

  setTheme = () => {
    const { theme } = this.props
    const { themeConfig } = this.state
    const updatedTheme = themeConfig.configure({ theme })
    this.setState({ theme: updatedTheme })
  }

  render() {
    return (
      <Provider value={{ theme: this.state.themeConfig.theme }}>
        {this.props.children}
      </Provider>
    )
  }
}

export { ThemeProvider }

export default Consumer