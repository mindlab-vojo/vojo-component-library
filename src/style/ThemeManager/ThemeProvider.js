import React, { Component, createContext } from 'react'
import PropTypes from 'prop-types'

import ThemeStore from './ThemeStore';

// Provider and Consumer instances
const { Provider, Consumer } = createContext();

class ThemeProvider extends Component {
  static propTypes = {
    children: PropTypes.node,
    theme: PropTypes.string
  }

  state = {
    themeConfig: new ThemeStore()
  }

  componentDidMount() {
    if (this.props.theme) this.setTheme()
  }

  componentDidUpdate(prevProps) {
    if (this.props.theme !== prevProps.theme) this.setTheme()
  }

  setTheme = () => {
    const { theme } = this.props
    const { themeConfig } = this.state
    const updatedTheme = themeConfig.configure({ theme })
    this.setState({ theme: updatedTheme })
  }

  getThemeFont = () => {
    const { themeConfig } = this.state
    return themeConfig.font
  }

  render() {

    const GlobalFont = this.getThemeFont()

    return (
      <Provider value={{ theme: this.state.themeConfig.theme }}>
        <GlobalFont/>
        {this.props.children}
      </Provider>
    )
  }
}

export { ThemeProvider }

export default Consumer