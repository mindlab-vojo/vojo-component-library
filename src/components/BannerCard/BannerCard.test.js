import React from 'react';
import { render } from '@testing-library/react';

import BannerCard from './';
import { ThemeProvider } from './../../style/ThemeManager/ThemeProvider';

test('Test BannerCard component render', () => {
  const fakeFn = jest.fn()
  const component = render(
    <ThemeProvider>
      <BannerCard 
        description="Entregador Mágico"
        id="card-id"
        onClickCard={fakeFn}
        title="Entregador Mágico"
      />
    </ThemeProvider>
  );
  expect(component).toMatchSnapshot()
});
