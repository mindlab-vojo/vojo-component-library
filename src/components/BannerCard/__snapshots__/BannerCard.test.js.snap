// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test BannerCard component render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-gKsewC kKxGrg"
        id="BannerCard-card-id"
      >
        <div
          class="sc-bdfBwQ beCqgx BannerCard"
        >
          <div
            class="sc-dlfnbm dPJlIz BannerCard__LeftContent"
          >
            <div
              class="sc-gsTCUz cOJuHC BannerCard__LeftContent__Image"
            >
              <p
                class="sc-iBPRYJ iRPfyL"
              >
                ?
              </p>
            </div>
          </div>
          <div
            class="sc-hKgILt fjzqPh BannerCard__RightContent"
          >
            <div
              class="sc-jSgupP bAEpIu BannerCard__RightContent__Description"
            >
              <p
                class="sc-fubCfw chsKBR"
              >
                Entregador M치gico
              </p>
            </div>
            <div
              class="sc-eCssSg jvctCF BannerCard__RightContent__Title"
            >
              <p
                class="sc-pFZIQ gNUGul"
              >
                Entregador M치gico
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gKsewC kKxGrg"
      id="BannerCard-card-id"
    >
      <div
        class="sc-bdfBwQ beCqgx BannerCard"
      >
        <div
          class="sc-dlfnbm dPJlIz BannerCard__LeftContent"
        >
          <div
            class="sc-gsTCUz cOJuHC BannerCard__LeftContent__Image"
          >
            <p
              class="sc-iBPRYJ iRPfyL"
            >
              ?
            </p>
          </div>
        </div>
        <div
          class="sc-hKgILt fjzqPh BannerCard__RightContent"
        >
          <div
            class="sc-jSgupP bAEpIu BannerCard__RightContent__Description"
          >
            <p
              class="sc-fubCfw chsKBR"
            >
              Entregador M치gico
            </p>
          </div>
          <div
            class="sc-eCssSg jvctCF BannerCard__RightContent__Title"
          >
            <p
              class="sc-pFZIQ gNUGul"
            >
              Entregador M치gico
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
